RFCs/0070-Fn::Size.md

```diff
@@ -0,0 +1,99 @@
+# RFC for Fn::Size
+
+* **Original Author(s):**: @juegong2
+* **Tracking Issue**: https://github.com/aws-cloudformation/cfn-language-discussion/issues/70
+
+# Summary
+
+We will support an intrinsic function called Fn::Size that will return the number of elements in a given list.
+
+# Motivation
+
+CloudFormation users may want to get the number of elements in a given list to set certain resource parameters or generate conditions. It may also help to check if index is out of bound for !Select function. This RFC proposes to support the use case of getting the number of elements of a given list.
+
+* Reference:
+  * https://github.com/aws-cloudformation/cfn-language-discussion/issues/61
+  * https://github.com/aws-cloudformation/cfn-language-discussion/issues/37
+
+# Example(s)
+
+Use `Ref` on a Parameter of type `CommaDelimitedList` or `List<Number>`
+
+```json
+{
+  "Parameters": {
+    "InstanceTypes": {
+      "Type": "CommaDelimitedList"
+    },
+    "NumberList": {
+      "Type": "List<Number>"
+    }
+  }
+  "Conditions": {
+    "Has3InstanceTypes": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": {"Ref": "InstanceTypes"}
+        },
+        3
+      ]
+    },
+    "Has2NumbersInList": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": {"Ref": "NumberList"}
+        },
+        2
+      ]
+    }
+  }
+}
+```
+
+Use a hardcoded list
+```json
+{
+  "Conditions": {
+    "Has3InstanceTypes": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": [ "m5.large", "m5.xlarge" ]
+        },
+        3
+      ]
+    }
+  }
+}
+```
+
+# Limitation
+
+`Fn::Size` does not support reference to resource properties. It only supports reference to Parameters.
```

Comment:
I'm a little confused by this statement. Are there any resource types with return values of type list? I'm pretty sure they're all strings... Should be more precise and say `Fn::Size` will not work with lists that are dynamically generated at provisioning time. For example, assuming MyBucket is the logical id of a `AWS::S3::Bucket` resource, the following is not supported:

```yaml
Fn::Size:
  Fn::Split: [':', !GetAtt MyBucket.Arn]
```

because the ARN of MyBucket is not known until during provisioning.